packages/rocketchat-federation/server/methods/federationSearchUser.js

```diff
@@ -0,0 +1,34 @@
+import { Meteor } from 'meteor/meteor';
+import { Federation } from '../federation';
+
+export function findFederatedUser(email) {
+	const [username, domain] = email.split('@');
+
+	const searchParameters = { domain };
+
+	// Easter egg: adding a `~` prefix will make
+	// the search happen by username and not email
+	if (email.indexOf('~') === 0) {
```

Comment:
So is this what we are going to make required to talked to users of other instances that don't use domains?

Reply:
@alansikora could we make this a toggle on the otherside?  Let it search based on email if they have setup their corporate domain.  Otherwise search by username?

Reply:
Alan is already working on a new version that will search for both (email + username@domain).. we decided to go this path after talking to @engelgabriel .. just waiting Alan to push the code so I can test =)

Reply:
Ok sweet!  Just wanted to make sure this didn’t get lost and saw this when checking to see how implemented 😁